### ISSUE TITLE
ValueError: Extraction of MNIST\raw\train-images-idx3-ubyte not supported

### ISSUE THREAD

**Issue by Wemmons831** [pytorch/vision#3554]
Hey, so I am trying to use the MNIST dataset by doing
`train = datasets.MNIST("", train =True, download = True, transform = transforms.Compose([transforms.ToTensor()]))
test = datasets.MNIST("", train =False, download = True, transform = transforms.Compose([transforms.ToTensor()]))`
however, I got a 503 error because the link for the download is down. I know you guys have no control over this so I downloaded the files off of another website. I know there is probably a different way to do this but I couldn't find the file path to put them in so I just hosted the files on a flask website and changed the URL in mnist.py to the URL I was hosting the files on. This worked fine and it downloaded all the files but now when I run it I get the following error
`ValueError: Extraction of MNIST\raw\train-images-idx3-ubyte not supported`
And I tried unzipping the files and placing them in a bunch of different folders but that didn't seem to do anything so I was wonder if anyone knew how to fix this

**Comment by CanIyu** [comment]
You can solve this below command ! 
But It took to download in about 10 min.

`train_val = datasets.MNIST('./', train=True, download=True, transform=transform)
test = datasets.MNIST('./', train=False, download=True, transform=transform)`

**Comment by Wemmons831** [comment]
I still get the same error could it be that I got the wrong files cause I am just using the latest ones from the website with Wayback machine

**Comment by Wemmons831** [comment]
> You can solve this below command !
> But It took to download in about 10 min.
> 
> `train_val = datasets.MNIST('./', train=True, download=True, transform=transform)`
> `test = datasets.MNIST('./', train=False, download=True, transform=transform)`
I still get the same error could it be that I got the wrong files cause I am just using the latest ones from the website with Wayback machine

**Comment by Wemmons831** [comment]
I still get the same error could it be that I got the wrong files cause I am just using the latest ones from the website with Wayback machine

**Comment by Wemmons831** [comment]
It is trying to extract a file that has already been extracted from the gzip

**Comment by ptrblck** [comment]
Cross-posting:
It seems the MNIST server might be down (or is dying often):
https://discuss.pytorch.org/t/mnist-server-down/114433

http://yann.lecun.com/exdb/mnist/ shows `Service Unavailable` randomly.

**Comment by pmeier** [comment]
> I know you guys have no control over this so

We got permission to host the dataset ourselves and provide the download in case the original server breaks again. See #3544 for details. 

> `ValueError: Extraction of MNIST\raw\train-images-idx3-ubyte not supported`

This happens, because right now we are using `datasets.utils.extract_archive` to also decompress files. Since you already have extracted the file it doesn't know what to do with it. I'm working on a proper fix for this in #3443 as we speak. 

For a temporary workaround you can do something like this to compress the files again.

```python
import gzip

for file in files:
    with open(file, "rb") as rfh, gzip.open(f"{file}.gz", "wb") as wfh:
        wfh.write(rfh.read())
```

**Comment by Avs163** [comment]
> You can solve this below command !
> But It took to download in about 10 min.
> 
> `train_val = datasets.MNIST('./', train=True, download=True, transform=transform)`
> `test = datasets.MNIST('./', train=False, download=True, transform=transform)`

Thanks man, this is working fine :)
